### PR TITLE
Correct websub signifiers

### PR DESCRIPTION
**Pull request by KaiTries**
This should solve issue #52 . 

The signifiers are now correctly located in the entity and the topic is now also correctly stated.

### CODE REVIEW

File: yggdrasil-utils/src/main/java/org/hyperagents/yggdrasil/utils/impl/RepresentationFactoryHMASImpl.java [hunk 1]
```diff
@@ -94,418 +104,444 @@ private Signifier webSubSignifier(final WebSubSpecs specs,
...
 
     // registerArtifact Signifier
     final Form registerArtifactForm = new Form.Builder(baseUri + "artifacts/")
-      .setMethodName(HttpMethod.POST.name())
-      .setContentType(CONTENT_TYPE_TURTLE)
-      .setIRIAsString(baseUri + "#registerArtifactForm")
-      .build();
-    final QualifiedValueSpecification registerArtifactInput = new QualifiedValueSpecification.Builder()
-      .addRequiredSemanticType(CORE.TERM.ARTIFACT.toString())
-      .setIRIAsString("http://example.org/artifact-rdf")
```
Comment (danaivach) [inline yggdrasil-utils/src/main/java/org/hyperagents/yggdrasil/utils/impl/RepresentationFactoryHMASImpl.java:235]: Consider replacing "http://example.org/artifact-rdf" with an IRI that uses the `baseUri`, e.g., `baseUri + "#artifact-rdf"`.

**Review by danaivach — CHANGES_REQUESTED** [review]

**Comment by danaivach** [comment]
@KaiTries, all looks good regarding issue https://github.com/Interactions-HSG/yggdrasil/issues/52.

If additional changes are applied, e.g., regarding the config, please provide a brief overview here before merging. Thanks!

File: README.md [hunk 2]
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
...
+        "template" : "org.hyperagents.yggdrasil.cartago.artifacts.CounterTD"
+      }
+    ]
+  }
+}
+```
+The "known-artifacts" key takes a JsonArray as its value. Each Json Object contains the class and template for an artifact.
+The class is then used as a key to identify the artifacts at runtime. The template is simply the java class of the wanted artifact.
+When a virtual artifact is in the "known-artifacts" array users are then able to create instances of this class by using the makeArtifact endpoint
+in a workspace (needs to be a workspace that has an underlying cartago workspace). Correctly specified artifacts can then be used to remotely
```
Comment (danaivach) [inline README.md:154]: cartago -> CArtAgO

File: README.md [hunk 3]
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
...
+    ]
+  }
+}
+```
+The "workspaces" key again takes as value a JsonArray where each JsonObject specifies a Workspace.
+If you have the environment enabled, a workspace only needs a name and every other variable is optional.
+If the environment is disabled the "representation" parameter is also mandatory.
+
+##### Environment disabled
+If the environment is disabled, we do not use Cartago and therefore provide less functionality. This also means
```
Comment (danaivach) [inline README.md:202]: cartago -> CArtAgO

File: README.md [hunk 4]
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
...
+    ]
+  }
+}
+```
+Optionally you can also define parent - child relationships between workspaces.
+
+#### Environment enabled
+If the environment is enabled, the given yggdrasil instance will provide more funcionality in interacting with the environment.
+This is reflected in the more extendable configuration file. It is still possible to specifiy a workspace with a name and representation.
+If you do it this way NO underlying cartago workspace will be created!
```
Comment (danaivach) [inline README.md:233]: cartago -> CArtAgO

File: README.md [hunk 5]
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
...
+
+##### workspaces
+The following configurations will all create an underlying cartago workspace instance.
```
Comment (danaivach) [inline README.md:236]: cartago -> CArtAgO

File: README.md [hunk 6]
```diff
@@ -79,6 +79,228 @@ Run and build the Yggdrasil image in the project directory context
...
+If instead the "class" key is specified then the application will check the "known-artifacts" array and if the "class" is
+present it will create an instance of the specified template class. It is possible to instantiate artifacts with parameters,
+these are specified with the optional "init-params" key. Additional metadata can again be specified through the "metadata" parameter.
+
+##### agents
+Enabling the enviroment allows the creation of body artifacts and the ability of agents to join and participate in the environment.
+An agent consists of a name which will be given to the body artifact, an agent-uri which should point to the WebID of the agent and a callback-uri
+which should point to an endpoint of the agent that is capable of handling the update messages sent from yggdrasil.
+
+Specifying an agent in a workspace will automatically create a body artifact in said workspace as well as join the agent in the cartago workspace.
```
Comment (danaivach) [inline README.md:296]: cartago -> CArtAgO

Comment (danaivach) [reply README.md:296]: etc.

**Comment by KaiTries** [comment]
implemented the comments by @danaivach . 

Since this branch built untop of my merge branch, there are some additional features that are part of this pull request:

- Complete reformatting such that the entire code base once again conforms to the checkstyle guidelines, refactored the code acording to pmd and checkBugs. 
- Renamed RdfStoreVerticle function "UpdateEntity" -> "ReplaceEntity" so that it is closer to its actual functionality
- Added new RdfStoreVerticle function "UpdateEntity" that actually updates a representation and does not replace already existing data. 
    - HttpHandler now utilizes this new function and when creating an entity through turtle, first a base representation is created and then the user sent representation is updated on top of that entity.
- Base Refactor of the configuration files for issue #56 
    - This came with some refactoring of the initialization code and the environment parser but does not cause any change in functionality
    - Added support for the "metadata" tag so that cartago entities can be initialized with additional metadata.
- Updated README with an overview of the config file structure

**Comment by KaiTries** [comment]
still TODO:

- update test ttl files again

**Comment by KaiTries** [comment]
Updated all test files and reformatted changes to be checkstyle conforming. 

Should be ready to merge

**Review by danaivach — APPROVED** [review]

**Review by andreiciortea — APPROVED** [review]
Thanks @KaiTries!

**Pull request merged**